draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,39 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function extension involved in the expression
+must be supported.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
+
+Implementations MUST NOT support function extensions that are not registered.
+Note: if an implementation supports a function extension that is not registered,
+the implementation will not conform to this specification until the function extension is
+registered.
```

Comment:
I think this is unnecessary.  It also implies that any custom functions will be proposed for registration, which is unlikely.  Mostly, I expect users will extend an implementation with some custom function for their use only and be satisfied with that without considering registration.

Reply:
Does this help?
```suggestion
Note: if an implementation supports a function extension that is not registered,
the implementation will not conform to this specification unless or until the function extension is
registered.
```

Reply:
I think the whole note is unnecessary.  I think it's generally implied that custom functionality outside of the scope of a spec.

Reply:
Ok, I deleted that note.